docs/reference/outlets.md

```diff
@@ -0,0 +1,179 @@
+| Plural | `[name]Outlets` | `Array<Controller>` | Returns the `Controller` instances of all `[name]` outlets
+| Singular | `[name]OutletElement` | `Element` | Returns the Controller `Element` of the first `[name]` outlet or throws an exception if none is present
+| Plural | `[name]OutletElements` | `Array<Element>` | Returns the Controller `Element`'s of all `[name]` outlets
+
+## Accessing controllers and elements
+
+Since you get back a `Controller` instance from the `[name]Outlet` and `[name]Outlets` properties you are also able to access the Values, Classes, Targets and all of the other properties and functions that controller instance defines:
+
+```js
+this.resultOutlet.idValue
+this.resultOutlet.imageTarget
+this.resultOutlet.activeClasses
+```
+
+You are also able to invoke any function the outlet controller may define:
+
+```js
+// result_controller.js
+
+export default class extends Controller {
+  markAsSelected(event) {
+    // ...
+  }
+}
+
+// search_controller.js
+
+export default class extends Controller {
+  static outlets = [ "result" ]
+
+  selectAll(event) {
+    this.resultOutlets.forEach(result => result.markAsSelected(event))
+  }
+}
+```
+
+Similarly with the Outlet Element, it allows you to call any function or property on [`Element`](https://developer.mozilla.org/en-US/docs/Web/API/Element):
+
+```js
+this.resultOutletElement.dataset.value
+this.resultOutletElement.getAttribute("id")
+this.resultOutletElements.map(result => result.hasAttribute("selected"))
+```
+
+## Outlet callbacks
+
+Outlet callbacks are specially named functions called by Stimulus to let you respond to whenever an outlet is added or removed from the page.
+
+To observe outlet changes, define a function named `[name]OutletConnected()` or `[name]OutletDisconnected()`.
+
+```js
+// search_controller.js
+
+export default class extends Controller {
+  static outlets = [ "result" ]
+
+  resultOutletConnected(outlet, element) {
+    // ...
+  }
+
+  resultOutletDisconnected(outlet, element) {
+    // ...
+  }
+}
+```
+
+### Outlets are assumed to be present
+
+When you access an outlet property in a controller, you assert that at least one corresponding outlet is present. If the declaration is missing and no matching outlet is found Stimulus will throw an exception:
+
+```html
+Missing outlet element "result" for "search" controller
+```
+
+### Optional outlets
+
+If an outlet is optional or you want to assert that at least outlet is present, you must first check the presence of the outlet using the existential property:
+
+```js
+if (this.hasResultOutlet) {
+  this.resultOutlet.safelyCallSomethingOnTheOutlet()
+}
+```
+
+### Referencing non-controller elements
+
+Stimulus will throw an exception if you try to declare an element as an outlet which doesn't have the corresponding `data-controller` and identifier on it:
```

Comment:
```suggestion
Stimulus will throw an exception if you try to declare an element as an outlet which doesn't have a corresponding `data-controller` and identifier on it:
```

I'm not clear here - does the name of the Outlet's controller have to match the `static outlets = ['something']` name?

Reply:
It needs to match the name of the outlets, yes. The `outlets` array holds references to other controller identifiers.